artemis-cli/src/main/java/org/apache/activemq/artemis/cli/commands/Upgrade.java

```diff
@@ -0,0 +1,318 @@
+            OutputStream outputStream = new FileOutputStream(newLogging);
+            outputStream.write(inputStream.readAllBytes());
+            inputStream.close();
+         }
+      }
+
+
+
+      File bootstrap = new File(etc, "bootstrap.xml");
+      if (bootstrap.exists()) {
+         File bootstrapCopy = new File(bkpFolder, "bootstrap.xml");
+         Files.copy(bootstrap.toPath(), bootstrapCopy.toPath(), StandardCopyOption.REPLACE_EXISTING);
+         System.out.println("Converting bootstrap.xml. Original file backed up at " + bkpFolder.getAbsolutePath());
+
+         upgrade(bootstrapCopy, bootstrap, (input, iterator, out) -> {
+            if (input.trim().startsWith("<server configuration=")) {
+               return "   <server configuration=\"" + etc.toURI() + "/broker.xml" + "\"/>";
+            }
+
+            return input;
+         });
+      } else {
+         System.out.println("Could not find bootstrap.xml");
+      }
+   }
+
+
+
+   private void upgradeBootstratp(File bkpFolder, File etc, File instance) throws Exception {
+      File bootstrap = new File(etc, "bootstrap.xml");
+      if (bootstrap.exists()) {
+         File bootstrapCopy = new File(bkpFolder, "bootstrap.xml");
+         Files.copy(bootstrap.toPath(), bootstrapCopy.toPath(), StandardCopyOption.REPLACE_EXISTING);
+         System.out.println("Converting bootstrap.xml. Original file backed up at " + bkpFolder.getAbsolutePath());
+
+         upgrade(bootstrapCopy, bootstrap, (input, iterator, out) -> {
+            if (input.trim().startsWith("<server configuration=")) {
+               return "   <server configuration=\"" + etc.toURI() + "/broker.xml" + "\"/>";
+            }
+
+            return input;
+         });
+      } else {
+         System.out.println("Could not find bootstrap.xml");
+      }
+   }
+
+
+   private void upgradeArtemisProfile(File bkpFolder, File etc, File instance) throws Exception {
+      File artemisProfile = new File(etc, "artemis.profile");
+      if (artemisProfile.exists()) {
+         File artemisProfileCopy = new File(bkpFolder, "artemis.profile");
+         Files.copy(artemisProfile.toPath(), artemisProfileCopy.toPath(), StandardCopyOption.REPLACE_EXISTING);
+         System.out.println("Converting artemis profile. Original file backed up at " + bkpFolder.getAbsolutePath());
+
+         upgrade(artemisProfileCopy, artemisProfile, (input, iterator, out) -> {
+
+            if (input.startsWith("ARTEMIS_HOME=")) {
+               return "ARTEMIS_HOME='" + getHome().getAbsolutePath() + "'";
+            }
+
+            if (input.startsWith("ARTEMIS_INSTANCE=")) {
+               return "ARTEMIS_INSTANCE='" + instance.getAbsolutePath() + "'";
+            }
+
+            if (input.startsWith("ARTEMIS_DATA_DIR=")) {
+               // TODO: this variable is not really used, we should get rid of it
+               return "ARTEMIS_DATA_DIR='" + instance.getAbsolutePath() + "/data'";
```

Comment:
The data directory can be configured on create. Why are we modifying these directories on upgrade anyway?

e.g. in one of our instances
```
ARTEMIS_HOME='/opt/apache-artemis-2.22.0'
ARTEMIS_INSTANCE='/rave/ba_artemis/instances/msg-sms1'
ARTEMIS_DATA_DIR='/data/ba_artemis/msg-sms1'
ARTEMIS_ETC_DIR='/rave/ba_artemis/instances/msg-sms1/etc'
ARTEMIS_OOME_DUMP='/logs/ba_artemis/oom_dump.hprof'
```

Reply:
ok.. fair enough...


I will make updating the instance an optional argument. I have to recast the instance on my upgrading test.. and I guess it would make sense if you are relocating the instance to a new directory.


Making it optional would fix that.

Reply:
Alternately add all the same directory arguments to upgrade as create, but yes your way makes sense for the common case of in place upgrades.